### PR TITLE
Implement example promotional-collections

### PR DESCRIPTION
**Pull request by nlisgo**

### CODE REVIEW

File: data/promotional-collections/highlights-japan.json [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        {
+            "type": "paragraph",
+            "text": "eLife has published papers on many tropical diseases, including malaria, Ebola, leishmaniases, Dengue and African sleeping sickness. The articles below have been selected by eLife editors to give a flavour of the breadth of research on tropical diseases published by the journal."
+        }
+    ],
+    "published": "2020-01-20T11:19:26Z",
+    "image": {
+        "banner": {
+            "alt": "",
+            "uri": "https://iiif.elifesciences.org/journal-cms/collection%2F2015-09%2F2d2e35c4-tropical-disease-elife-collections.png",
```
Comment (giorgiosironi) [inline data/promotional-collections/highlights-japan.json:15]: no problem with the images being real URLs as it probably matches the rest of this project

File: data/promotional-collections/highlights-japan.json [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+                    "width": 4256,
+                    "height": 2832
+                }
+            }
+        }
+    ],
+    "content": [
+        {
+            "type": "blog-article",
+            "id": "359325",
```
Comment (giorgiosironi) [inline data/promotional-collections/highlights-japan.json:90]: are the snippets in these two samples present in their full form in the `api-dummy` already? e.g. `/blog-articles/359325`. From the familiar titles, I think so.

File: src/bootstrap.php [hunk 3]
```diff
@@ -1812,6 +1829,89 @@ function (Accept $type, string $id) use ($app) {
...
+    }
+
+    $promotionalCollections = array_slice($promotionalCollections, ($page * $perPage) - $perPage, $perPage);
+
+    if (0 === count($promotionalCollections) && $page > 1) {
+        throw new NotFoundHttpException('No page '.$page);
+    }
+
+    foreach ($promotionalCollections as $i => $promotionalCollection) {
+        unset($promotionalCollection['editors']);
```
Comment (giorgiosironi) [inline src/bootstrap.php:1878]: the most relevant bit with the properties that are different from the `collection` ones

File: src/bootstrap.php [hunk 4]
```diff
@@ -1812,6 +1829,89 @@ function (Accept $type, string $id) use ($app) {
...
+            $promotionalCollectionsSubjects = array_map(function (array $subject) {
+                return $subject['id'];
+            }, $promotionalCollection['subjects'] ?? []);
+
+            return count(array_intersect($subjects, $promotionalCollectionsSubjects));
+        });
+    }
+
+    if (false === empty($containing)) {
+        $promotionalCollections = array_filter($promotionalCollections, function ($promotionalCollection) use ($containing) {
```
Comment (giorgiosironi) [inline src/bootstrap.php:1851]: I assume this only applies to `content` and not to:
- `relatedContent` (as it's kind of a suggestion for other content, but not contained in here)
- `podcastEpisodes` (which are managed separately)

Comment (nlisgo) [reply src/bootstrap.php:1851]: This is the same way that it is implemented with collections.

Comment (giorgiosironi) [reply src/bootstrap.php:1851]: Yes, just stating it out loud to see if I was mistaken.

File: src/bootstrap.php [hunk 5]
```diff
@@ -580,8 +597,8 @@ function (Request $request, string $number) use ($app) {
 
         $latestVersion = count($app['articles'][$number]['versions']);
 
-        $subRequest = Request::create('/articles/'.$number.'/versions/'.$latestVersion, 'GET', array(),
-            $request->cookies->all(), array(), $request->server->all());
+        $subRequest = Request::create('/articles/'.$number.'/versions/'.$latestVersion, 'GET', [],
+            $request->cookies->all(), [], $request->server->all());
```
Comment (nlisgo) [inline src/bootstrap.php:601]: This is not related but was cleaned up automatically. Will create a new PR just for this.

**Comment by nlisgo** [comment]
Will merge this in after https://github.com/elifesciences/api-dummy/pull/158

**Pull request merged**